app/models/concerns/user_select_option.rb

```
@@ -0,0 +1,53 @@
+module UserSelectOption
+  extend ActiveSupport::Concern
+
+  module ClassMethods
+    def all_options_for(version:)
+      select_options = {}
+      allowed_select_options.each_key do |option|
+        select_options[option] = allowed_options_for(option, version: version)
+      end
+
+      select_options
+    end
+
+    def allowed_options_for(option, version:)
+      return nil if allowed_select_options[option].nil? || version == 0
```

couple questions to consider:
1) Should you `raise` rather than returning `nil`?  I don't have a fixed PoV (i can go either way), but just want to consider
2) What if you pass `version` as `nil`. should that `raise` or return `nil`?